scripts/version.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+# TODO: where to extract the latest version of the tools from?
+
+# Source: https://semver.org/
+# Using a simplified version of semantic versioning regex pattern, which is bash compatible
+SEM_VER_REGEX="^([0-9]+)\.([0-9]+)\.([0-9]+)(-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$"
+
+# Set default values for variables
+KUBECTL_VERSION_DEFAULT="1.31.3"
+GO_VERSION_DEFAULT="1.23.3"
+K3D_VERSION_DEFAULT="5.6.0"
+DOCKER_VERSION_DEFAULT="27.3.1"
+ISTIOCTL_VERSION_DEFAULT="1.24.1"
+
+versioning_error=false
+# Check if required tools are installed
+if ! command -v kubectl &> /dev/null; then
+  echo "kubectl is not installed. Please install kubectl."
+  versioning_error=true
+fi
+
+if ! command -v go &> /dev/null; then
+  echo "Go is not installed. Please install Go."
+  versioning_error=true
+fi
+
+if ! command -v k3d &> /dev/null; then
+  echo "k3d is not installed. Please install k3d."
+  versioning_error=true
+fi
+
+if ! command -v docker &> /dev/null; then
+  echo "Docker is not installed. Please install Docker."
+  versioning_error=true
+fi
+
+if ! command -v istioctl &> /dev/null; then
+  echo "istioctl is not installed. Please install istioctl."
+  versioning_error=true
+fi
+
+if $versioning_error; then
+  exit 1
+fi
+
+# Versions installed on current system
+KUBECTL_VERSION_INSTALLED=$(kubectl version --client | head -n1 | awk '{print $3}' | sed 's/v//')
```

Comment:
I am getting
```
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Invalid kubectl version: ersion.Info{Major:"1",
```

Reply:
I tried to replicate this warning message with multiple versions of `kubectl`, but somehow couldn't get it. Anyway, I added another safeguarding `grep` statement. Hopefully it should mitigate this problem.